### ISSUE TITLE
Terraform 0.9.1 can't apply plan from a file with S3 backend

### ISSUE THREAD

**Issue by rafaelmagu** [hashicorp/terraform#12871]
It appears that Terraform is generating invalid plan files when `-out=` is used.

### Terraform Version
0.9.1

### Affected Resource(s)
Core Terraform.

### Expected Behavior
Expected to see `terraform apply file.out` apply any pending changes generated by `terraform plan -out=file.out`

### Actual Behavior
```
Failed to load backend: The plan file contained both a legacy remote state and backend configuration.
This is not allowed. Please recreate the plan file with the latest version of
Terraform.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `rm file.out` (just in case it exists currently)
2. `terraform plan -out=file.out`
3. `terraform plan file.out`

### Current Mitigation
I'm applying changes using `-target=` statements.

**Comment by mubeta06** [comment]
Yeah I am seeing this too after the upgrade.

**Comment by mitchellh** [comment]
I found the issue. The migration code doesn't clear the remote section remotely (it clears it locally in what is called the "data" state which only has the backend configuration). I will fix this early in the AM tomorrow.

**Comment by tacchino** [comment]
Same thing happens with an Atlas backend

**Comment by mitchellh** [comment]
Yes this is a generic problem with any legacy remote state. Plan on working on a fix here shortly.

**Comment by mitchellh** [comment]
Fix in attached PR, will be part of 0.9.2.
